### PR TITLE
Add netlify config for cloud provider azure

### PR DESCRIPTION
**Pull request by MartinForReal**
Signed-off-by: MartinForReal <fanshangxiang@gmail.com>

#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
add netlify config as requested by admin
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by MartinForReal** [comment]
related: https://github.com/kubernetes/org/issues/3268#issuecomment-1155121919

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# HTTP-to-HTTPS rules
+[[redirects]]
+    from = "http://kubernetes-sigs-cloud-provide-azure.netlify.app/*"
+    to = "https://cloud-provider-azure.sigs.k8s.io/:splat"
+    status = 301
+    force = true
+
+[[redirects]]
+    from = "http://main--kubernetes-sigs-cluster-api.netlify.com/*"
```
Comment (jwtty) [inline netlify.toml:26]: Why do we need this one?

Comment (MartinForReal) [reply netlify.toml]: updated. Thanks!

**Comment by jwtty** [comment]
Can we add a bit more about why we need this?
Otherwise /lgtm

**Comment by MartinForReal** [comment]
netlify is a popular document publishing service and is widely used among kubernetes projects. ref: https://github.com/kubernetes/org/search?q=netlify&type=issues

And I opened a issue here to create a new site for cloud provider azure https://github.com/kubernetes/org/issues/3268#issuecomment-1155121919

**Pull request merged**